src/contrib/torch/tvm_module_wrapper/RuntimeModuleWrapperTVM.cc

```diff
@@ -0,0 +1,179 @@
+  fwrite(bytes.data(), sizeof(u_char), length, pFile.get());
+  fflush(pFile.get());
+
+  std::string load_f_name = "runtime.module.loadfile_so";
+  const PackedFunc* f = runtime::Registry::Get(load_f_name);
+  ICHECK(f != nullptr) << "Loader for `.so` files is not registered,"
+                       << " resolved to (" << load_f_name << ") in the global registry."
+                       << "Ensure that you have loaded the correct runtime code, and"
+                       << "that you are on the correct hardware architecture.";
+
+  tvm::runtime::Module ret = (*f)(file_name, "");
+
+  return ret;
+}
+
+tvm::Device getDeviceInfo(DLManagedTensor* input_device) {
+  return {.device_type = input_device->dl_tensor.device.device_type,
+          .device_id = input_device->dl_tensor.device.device_id};
+}
+
+TVM_REGISTER_GLOBAL("tvmtorch.save_runtime_mod").set_body_typed([](tvm::runtime::Module mod) {
+  ThreadLocalStore::ThreadLocal()->mod = mod;
+});
+
+}  // namespace contrib
+}  // namespace tvm
+
+extern "C" {
+
+struct TVMContribTorchRuntimeModule {
+  tvm::runtime::Module mod;
+
+  explicit TVMContribTorchRuntimeModule(tvm::runtime::Module mod) : mod(mod) {}
+};
+
+TVMContribTorchRuntimeModule* tvm_contrib_torch_get_last_saved_runtime_module() {
+  return new TVMContribTorchRuntimeModule(ThreadLocalStore::ThreadLocal()->mod);
+}
+
+void tvm_contrib_torch_operator_module_forward(TVMContribTorchRuntimeModule* runtime_module,
+                                               DLPackTensorExt* inputs, size_t input_size) {
+  tvm::runtime::PackedFunc run = runtime_module->mod.GetFunction("__tvm_main__");
+
+  std::vector<TVMValue> tvm_values(input_size);
+  std::vector<int> tvm_type_codes(input_size);
+  tvm::runtime::TVMArgsSetter setter(tvm_values.data(), tvm_type_codes.data());
+  for (int k = 0; k < input_size; ++k) {
+    setter(k, &inputs[k].dl_managed_tensor->dl_tensor);
+  }
+  run.CallPacked(tvm::runtime::TVMArgs(tvm_values.data(), tvm_type_codes.data(), input_size),
+                 nullptr);
+}
+
+int64_t tvm_contrib_torch_graph_executor_module_forward(TVMContribTorchRuntimeModule* graph_module,
+                                                        DLPackTensorExt* inputs, size_t input_size,
+                                                        DLPackTensorExt** outputs) {
+  tvm::runtime::PackedFunc built_module = graph_module->mod.GetFunction("default");
+  auto device_info = tvm::contrib::getDeviceInfo(inputs[0].dl_managed_tensor);
+  tvm::runtime::Module runtime_module = built_module(device_info);
+  tvm::runtime::PackedFunc run = runtime_module.GetFunction("run");
+  tvm::runtime::PackedFunc set_input = runtime_module.GetFunction("set_input");
+  tvm::runtime::PackedFunc get_output = runtime_module.GetFunction("get_output");
+  tvm::runtime::PackedFunc get_num_outputs = runtime_module.GetFunction("get_num_outputs");
+
+  for (int k = 0; k < input_size; ++k) {
+    set_input(k, &inputs[k].dl_managed_tensor->dl_tensor);
```

Comment:
If the input's dtype is boolean, in `RuntimeModuleWrapperTorch.cc` it will be cast into uint8 and the `dl_tensor` here will have dtype uint8. Although the two test cases on boolean tensor pass, is this logic correct in general?